wpilibj/src/main/java/edu/wpi/first/wpilibj/Tracer.java

```diff
@@ -21,6 +24,14 @@ public class Tracer {
 
   private long m_lastEpochsPrintTime; // microseconds
   private long m_startTime; // microseconds
+  private boolean m_publishNT;
+  private NetworkTableInstance m_inst;
+  private String m_ntTopic;
+  private Map<String, IntegerPublisher> m_publisherCache;
```

Comment:
These need to be closed somewhere.

Reply:
Like AutoCloseable's close() or something else?

Reply:
Exactly.
How it should be done given none of these are currently AutoCloseable, I'm not sure.

Reply:
In close(), I'm closing all the publishers in the Map.